src/schema/schema_alter.c

```diff
@@ -250,9 +250,12 @@ __wt_schema_alter(WT_SESSION_IMPL *session,
     const char *uri, const char *newcfg[])
 {
 	WT_DECL_RET;
+	WT_SESSION_IMPL *int_session;
 
-	WT_RET(__wt_meta_track_on(session));
-	ret = __schema_alter(session, uri, newcfg);
-	WT_TRET(__wt_meta_track_off(session, true, ret != 0));
+	WT_RET(__wt_schema_internal_session(session, &int_session));
+	WT_ERR(__wt_meta_track_on(int_session));
```

Comment:
Why does alter need to use meta tracking and other operations don't? A comment would be useful since this is different to other usage patterns.

Reply:
The new session code didn't change/add the meta_tracking usage. It only changed a `WT_RET` or `WT_TRET` into a `WT_ERR` so we could release the session.  All the other operations do tracking, just in other functions.